File: movierecommendationsystem-backend/src/main/java/com/movierecommendationsystem/movierecommendationsystem_backend/dto/Media.java
package com.movierecommendationsystem.movierecommendationsystem_backend.dto;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.movierecommendationsystem.movierecommendationsystem_backend.entity.Genre;

public class Media {
    @JsonProperty("id")
    private Long id;

    @JsonProperty("overview")
    private String overview;

    @JsonProperty("poster_path")
    private String posterPath;

    @JsonProperty("backdrop_path")
    private String backdropPath;

    @JsonProperty("popularity")
    private Double popularity;

    @JsonProperty("vote_average")
    private Double voteAverage;

    @JsonProperty("genres")
    private List<Genre> genres;
}

File: movierecommendationsystem-frontend/src/app/model/media.ts
export class Media{
    constructor(
        public id : number,
        public overview : string,
        public poster_path : string,
        public backdrop_path : string,
        public popularity : number,
        public vote_average : number,
        public genres : Genre[] = []
    ){}
}

export class Genre{
    constructor(
        public id : number,
        public name : string
    ){}
}